lib/trollop.rb

```diff
@@ -344,42 +578,16 @@ def method_missing(m, *_args)
     vals
   end
 
-  def parse_date_parameter(param, arg) #:nodoc:
-    begin
-      require 'chronic'
-      time = Chronic.parse(param)
-    rescue LoadError
-      # chronic is not available
-    end
-    time ? Date.new(time.year, time.month, time.day) : Date.parse(param)
-  rescue ArgumentError
-    raise CommandlineError, "option '#{arg}' needs a date"
-  end
-
   ## Print the help message to +stream+.
   def educate(stream = $stdout)
     width # hack: calculate it now; otherwise we have to be careful not to
           # call this unless the cursor's at the beginning of a line.
     left = {}
     @specs.each do |name, spec|
-      left[name] =
-        (spec.short? ? "-#{spec.short}, " : "") + "--#{spec.long}" +
-        case spec.type
-        when :flag    then ""
-        when :int     then "=<i>"
-        when :ints    then "=<i+>"
-        when :string  then "=<s>"
-        when :strings then "=<s+>"
-        when :float   then "=<f>"
-        when :floats  then "=<f+>"
-        when :io      then "=<filename/uri>"
-        when :ios     then "=<filename/uri+>"
-        when :date    then "=<date>"
-        when :dates   then "=<date+>"
-        end +
-        (spec.flag? && spec.default ? ", --no-#{spec.long}" : "")
+      type_edu = spec.educate
+      left[name] = (spec.short? ? "-#{spec.short}, " : "") + "--#{spec.long}" + type_edu + (spec.flag? && spec.default ? ", --no-#{spec.long}" : "")
```

Comment:
every method has `spec.` -- can you move these into the spec?

Reply:
:+1: